### PR TITLE
Adds ability to prepare/initialize before running `set_code` benchmark

### LINKED ISSUE #2765: Fix benchmarks `set_code` for `frame_system` pallet for all SP runtimes
here was added benchmark for `set_code` https://github.com/paritytech/substrate/pull/13373
but it does not work with Cumulus runtimes which uses:

```
type OnSetCode = cumulus_pallet_parachain_system::ParachainSetCode<Self>;
```
because it ends here, where some data are expected
https://github.com/paritytech/cumulus/blob/master/pallets/parachain-system/src/lib.rs#L986-L1008

so the fix should be to add some `T::BenchmarkHelper::prepare_data()` to the `set_code` benchmark [here](https://github.com/paritytech/substrate/blob/master/frame/system/benchmarking/src/lib.rs#L51-L56)

something like:
```
        set_code {
		let runtime_blob = include_bytes!("../res/kitchensink_runtime.compact.compressed.wasm").to_vec();
                T::BenchmarkHelper::prepare_data();
	}: _(RawOrigin::Root, runtime_blob)
	verify {
		System::<T>::assert_last_event(frame_system::Event::<T>::CodeUpdated.into());
	}
```

and then configure `BenchmarkHelper` in every runtime (or close to ParachainSetCode)

Example of failed job: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/3035163
```
./target/production/polkadot-parachain benchmark pallet --chain=asset-hub-kusama-dev --execution=wasm --wasm-execution=compiled --pallet=frame_system --no-storage-info --no-median-slopes --no-min-squares '--extrinsic=*' --steps=50 --repeat=20 --json --header=./file_header.txt --output=./parachains/runtimes/assets/asset-hub-kusama/src/weights/
Error: Input("Benchmark frame_system::set_code failed: ValidationDataNotAvailable")
```

### PR DESCRIPTION
**Pull request by bkontur**
See detailed description here: https://github.com/paritytech/cumulus/issues/2765

Closes: https://github.com/paritytech/cumulus/issues/2765

Cumulus companion: https://github.com/paritytech/cumulus/pull/2766

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3046047

**Comment by bkontur** [comment]
bot bench $ pallet dev frame-system
bot bench $ pallet dev frame-system-benchmarking

**Review by bkchr — APPROVED** [review]

**Comment by bkontur** [comment]
bot bench $ pallet dev frame-system

**Review by acatangiu — APPROVED** [review]

File: frame/system/benchmarking/src/lib.rs [hunk 1]
```diff
@@ -30,7 +33,18 @@ use sp_std::{prelude::*, vec};
 mod mock;
 
 pub struct Pallet<T: Config>(System<T>);
-pub trait Config: frame_system::Config {}
+pub trait Config: frame_system::Config {
+	/// Adds ability to the Runtime to test against their sample code.
+	/// Default is `../res/kitchensink_runtime.compact.compressed.wasm`.
+	fn set_code_data() -> Vec<u8> {
+		include_bytes!("../res/kitchensink_runtime.compact.compressed.wasm").to_vec()
+	}
```
Comment (bkchr) [inline frame/system/benchmarking/src/lib.rs:41]: This is a  very very good idea! :)

**Comment by bkontur** [comment]
bot bench $ pallet dev frame-system

**Comment by ggwpez** [comment]
bot bench-vm $ pallet dev frame-system

**Review by ggwpez — APPROVED** [review]
Need to wait for the new bench cmd to finish, then its good to go :rocket:

**Comment by bkontur** [comment]
bot merge

**Pull request merged**